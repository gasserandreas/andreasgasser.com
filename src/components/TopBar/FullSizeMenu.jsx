import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Styles = {
  FullSizeMenu: styled.div(({ theme, show, top }) => `
  position: fixed;
  // top: ${top}px;
  top: 0px;
  bottom: 0px;
  right: 0px;
  left: 0px;
  background-color: ${theme.colors.backgroundInverse};
  z-index: 100;
  display: ${show ? 'flex' : 'none'}
  align-items: center;
  justify-content: center;
`),
};

const FullSizeMenu = ({ show, top, children }) => (
  <Styles.FullSizeMenu show={show} top={top}>
    {children}
  </Styles.FullSizeMenu>
);

FullSizeMenu.propTypes = {
  show: PropTypes.bool,
  top: PropTypes.number.isRequired,
  children: PropTypes.node.isRequired,
};

FullSizeMenu.defaultProps = {
  show: false,
};

export default FullSizeMenu;
