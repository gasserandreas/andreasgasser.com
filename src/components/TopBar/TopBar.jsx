import React from 'react';
import { Box } from 'rebass/styled-components';

const TopBar = () => <Box>TopBar</Box>;

TopBar.propTypes = {};

TopBar.defaultProps = {
  siteTitle: '',
};

export default TopBar;
