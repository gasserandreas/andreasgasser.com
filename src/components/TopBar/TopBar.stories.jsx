import React from 'react';

import TopBar, { TOP_BAR_FULL_HEIGHT } from './TopBar';

export default {
  title: 'TopBar',
  component: TopBar,
};

export const defaultTopBar = () => (
  <div style={{
    position: 'absolute', top: '0px', height: '250px', overflow: 'scroll', borderBottom: '1px solid',
  }}
  >
    <TopBar menuContent={<p>Hello Menu</p>} />
    <p style={{ paddingTop: TOP_BAR_FULL_HEIGHT }}>
      Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt
      {' '}
      ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo
      {' '}
      duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum
      {' '}
      dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod
      {' '}
      tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam
      {' '}
      et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata
      {' '}
      sanctus est Lorem ipsum dolor sit amet.
      {' '}
      Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt
      {' '}
      ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo
      {' '}
      duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum
      {' '}
      dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod
      {' '}
      tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam
      {' '}
      et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata
      {' '}
      sanctus est Lorem ipsum dolor sit amet.
    </p>
  </div>
);

defaultTopBar.story = {
  name: 'default',
};
