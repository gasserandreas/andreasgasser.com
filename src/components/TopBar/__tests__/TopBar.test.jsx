/* global withTheme */
import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import TopBar from '../TopBar';

jest.mock('../../hooks/useEventListener', () => ({
  __esModule: true, // this property makes it work
  default: jest.fn(),
}));

describe('components/TopBar', () => {
  const MENU_TEXT_TEST_ID = 'topbar-menu-item';
  const MAIL_TEXT_TEST_ID = 'topbar-mail-item';
  const MENU_TEST_ID = 'topbar-menu';

  const menuContentText = 'Menu';
  const menuContent = (
    <p data-testid={MENU_TEST_ID}>
      {menuContentText}
    </p>
  );

  const contentRef = <div>content</div>;

  beforeAll(() => {
    global.open = jest.fn();
  });

  afterAll(() => {
    global.open.mockRestore();
  });

  it('should render default TopBar', async () => {
    const { getByTestId, findByTestId } = render(
      withTheme(
        <TopBar menuContent={menuContent} contentRef={contentRef} />,
      ),
    );

    const menuItem = getByTestId(MENU_TEXT_TEST_ID);
    expect(menuItem).toHaveTextContent('Menu');

    const mailItem = getByTestId(MAIL_TEXT_TEST_ID);
    expect(mailItem).toHaveTextContent('Email');

    // test to open menu
    fireEvent.click(menuItem);
    const menu = await findByTestId(MENU_TEST_ID);

    expect(menu).toHaveTextContent(menuContentText);

    // check for mailto link
    fireEvent.click(mailItem);
    expect(global.open).toMatchSnapshot();
  });
});
