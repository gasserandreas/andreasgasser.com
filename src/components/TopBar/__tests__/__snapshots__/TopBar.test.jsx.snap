// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/TopBar should render default TopBar 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "mailto:undefined?subject=Question from my website",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
